### PR TITLE
Introduce optional before each value.

### PR DESCRIPTION
**Pull request by MatejNedic**
## :loudspeaker: Type of change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
Introduces #114  changes to a project.
@eddumelendez  could you please review this.

## :bulb: Motivation and Context
This would be an awesome feature to have for people to import parameters and secrets.
If change is accepted I will reflect docs.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify changes
- [ ] I updated reference documentation to reflect the change
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: spring-cloud-starter-aws-secrets-manager-config/src/test/java/io/awspring/cloud/autoconfigure/secretsmanager/AwsSecretsManagerConfigDataLocationResolverTest.java [hunk 1]
```diff
@@ -53,6 +53,17 @@ void testResolveProfileSpecificWithCustomPaths() {
 		assertThat(toContexts(locations)).containsExactly("/mypath1", "/mypath2", "/mypath3");
 	}
 
+	@Test
+	void testResolveProfileSpecificWithCustomPathsOptional() {
+		String location = "aws-secretsmanager:optional /mypath1;/mypath2;optional /mypath3";
```
Comment (maciejwalkowiak) [inline spring-cloud-starter-aws-secrets-manager-config/src/test/java/io/awspring/cloud/autoconfigure/secretsmanager/AwsSecretsManagerConfigDataLocationResolverTest.java:58]: Why a space? Isn't it idiomatic to support: `aws-secretsmanager:optional:/mypath1;/mypath2;optional:/mypath3`?

File: spring-cloud-starter-aws-secrets-manager-config/src/main/java/io/awspring/cloud/autoconfigure/secretsmanager/AwsSecretsManagerConfigDataLocationResolver.java [hunk 2]
```diff
@@ -86,23 +88,36 @@ public boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDat
...
 
-	private List<String> getCustomContexts(String keys) {
+	private Map<String, Boolean> getCustomContexts(String keys) {
+		String optionalString = "optional";
+		Map<String, Boolean> mapOfValuesWithOptional = new HashMap<>();
 		if (StringUtils.hasLength(keys)) {
-			return Arrays.asList(keys.split(";"));
+			List<String> listOfFields = Arrays.asList(keys.split(";"));
+			listOfFields.forEach(field -> {
+				if (field.length() > 8 && field.toLowerCase().substring(0, 8).equals(optionalString)) {
```
Comment (maciejwalkowiak) [inline spring-cloud-starter-aws-secrets-manager-config/src/main/java/io/awspring/cloud/autoconfigure/secretsmanager/AwsSecretsManagerConfigDataLocationResolver.java:112]: This gets too low level. I think we could have a dedicated class `CustomContext` that would take a string as an argument, and have methods like `isOptional`, `getName`. Then `getCustomContexts` would return just a list of these objects.

If its not clear enough, we can pair on this one.

**Review by maciejwalkowiak — CHANGES_REQUESTED** [review]
Thanks @MatejNedic. Few things I believe we should change.

File: spring-cloud-starter-aws-secrets-manager-config/src/test/java/io/awspring/cloud/autoconfigure/secretsmanager/AwsSecretsManagerConfigDataLocationResolverTest.java (hunk 1 above)
Comment (MatejNedic) [reply spring-cloud-starter-aws-secrets-manager-config/src/test/java/io/awspring/cloud/autoconfigure/secretsmanager/AwsSecretsManagerConfigDataLocationResolverTest.java]: Agree it is better to use `:`

**Comment by MatejNedic** [comment]
Closing PR since we can do something like:
spring.config.import= optional:aws-secretsmanager:/test/user;/test/pass,aws-secretsmanager:/test/url

**Pull request closed without merging**